SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslator.cpp

```diff
@@ -3,39 +3,63 @@
 #include "EngineClasses/SpatialVirtualWorkerTranslator.h"
 #include "EngineClasses/SpatialNetDriver.h"
 #include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/SpatialReceiver.h"
+#include "Interop/SpatialStaticComponentView.h"
 #include "SpatialConstants.h"
 #include "Utils/SchemaUtils.h"
 
 DEFINE_LOG_CATEGORY(LogSpatialVirtualWorkerTranslator);
 
 SpatialVirtualWorkerTranslator::SpatialVirtualWorkerTranslator()
 	: NetDriver(nullptr)
-{
-}
+	, bWorkerEntityQueryInFlight(false)
+	, bIsReady(false)
+{}
 
 void SpatialVirtualWorkerTranslator::Init(USpatialNetDriver* InNetDriver)
 {
 	NetDriver = InNetDriver;
+	// If this is being run from tests, NetDriver will be null.
+	WorkerId = (NetDriver != nullptr) ? NetDriver->Connection->GetWorkerId() : "InvalidWorkerId";
```

Comment:
Trying to work out how to get rid of net driver.
It seems like it's needed for access to the view, sender, receiver, and the worker name.
The worker name and view feel like they could be passed in here.
For the sender and receiver we could pass them into the functions that use them, or pass the net driver in as a pointer there and capture it in the delegate.
Although it feels like we should really have non-uobjects controlling the worker connection interface and pass those in. But that feels like its out of scope here.

But would it be worth doing that to prevent Init(nullptr) in the tests?

Reply:
Discussed offline. We need to do this as an UnrealGDK project for sure, but it's outside the scope of this PR. This does highlight the need for a Fake Connection/NetDriver/whatever for testing, however.